Userland/Libraries/LibGL/SoftwareGLContext.h

```diff
@@ -138,6 +139,8 @@ class SoftwareGLContext : public GLContext {
     // Texture objects
     TextureNameAllocator m_name_allocator;
     HashMap<GLuint, RefPtr<Texture>> m_allocated_textures;
+    Array<TextureUnit, 32> m_texture_units;
+    OwnPtr<TextureUnit> m_active_unit { &m_texture_units[0] };
```

Comment:
The ownership model here is quite odd, who is supposed to own the `TextureUnit`, `m_texture_units` or `m_active_unit`?
As this is, the `OwnPtr` will call `free()` on the pointer (which will fail) and ignore the failure every time you change this active unit.

Reply:
Yeah I think either a raw pointer or an index into the array makes sense here.

Reply:
Well...fwiw this won't compile after #7602.

Reply:
I like the idea of a raw pointer here.

Reply:
A raw pointer is perfectly fine for this : )

Reply:
> 
> 
> I like the idea of a raw pointer here.

IMO an index would make more sense. Then we can reuse this for `glGet()` with `GL_ACTIVE_TEXTURE`